components/layout_2020/table/layout.rs

```diff
@@ -1921,40 +1921,34 @@ fn get_size_percentage_contribution_from_style(
     }
 }
 
-fn get_sizes_from_style(
+fn get_outer_sizes_from_style(
     style: &Arc<ComputedValues>,
     writing_mode: WritingMode,
+    padding_border_sums: &LogicalVec2<Au>,
 ) -> (LogicalVec2<Au>, LogicalVec2<Au>, LogicalVec2<Au>) {
-    let get_max_size_for_axis = |size: Option<&ComputedLengthPercentage>| {
-        size.and_then(|length_percentage| length_percentage.to_length())
-            .map_or(MAX_AU, Au::from)
+    let box_sizing = style.get_position().box_sizing;
+    let outer_size = |size: LogicalVec2<Au>| match box_sizing {
+        BoxSizing::ContentBox => &size + padding_border_sums,
+        BoxSizing::BorderBox => LogicalVec2 {
+            inline: size.inline.max(padding_border_sums.inline),
+            block: size.block.max(padding_border_sums.block),
+        },
     };
-
-    let max_size = style.max_box_size(writing_mode);
-    let max_size = LogicalVec2 {
-        inline: get_max_size_for_axis(max_size.inline),
-        block: get_max_size_for_axis(max_size.block),
-    };
-
-    let get_size_for_axis = |size: LengthPercentageOrAuto<'_>| {
+    let get_size_for_axis = |size: &LengthPercentageOrAuto<'_>| {
```

Comment:
This is no longer used on a per-axis basis, so perhaps this can just be `get_size`.